app/controllers/products_controller.rb

```diff
@@ -3,4 +3,20 @@ def index
     @pagy, @products = pagy Product.children_products,
                             items: Settings.number.digits_6
   end
+
+  def show
+    @product = Product.find_by id: params[:id]
+    if @product
+      @product_details = ProductDetail.product_size(@product.id).sizeM_XL
```

Comment:
```suggestion
      @product_details = ProductDetail.product_size(@product.id).sizeM_XL
      @product_details = @product.product_detail.sizeM_XL
```  

Như này nhìn ổn hơn nhỉ, với cả đặt tên scope clear hơn đi cậu, product_size thì nên truyền size vào chứ sao lại truyền id, sizeM_XL dùng để sort mà sao lại đặt tên như vậy nhìn vào sẽ k rõ mục đích nha